bftengine/src/bftengine/ReadOnlyReplica.cpp

```diff
@@ -128,4 +137,81 @@ void ReadOnlyReplica::onMessage<CheckpointMsg>(CheckpointMsg *msg) {
   }
 }
 
+template <>
+void ReadOnlyReplica::onMessage<ClientRequestMsg>(ClientRequestMsg *m) {
+  const NodeIdType senderId = m->senderId();
+  const NodeIdType clientId = m->clientProxyId();
+  const bool reconfig_flag = (m->flags() & MsgFlag::RECONFIG_FLAG) != 0;
+  const ReqId reqSeqNum = m->requestSeqNum();
+  const uint8_t flags = m->flags();
+
+  SCOPED_MDC_CID(m->getCid());
+  LOG_DEBUG(MSGS, KVLOG(clientId, reqSeqNum, senderId) << " flags: " << std::bitset<8>(flags));
+
+  const auto &span_context = m->spanContext<std::remove_pointer<decltype(m)>::type>();
```

Comment:
Nit: Since we are in the `ClientRequestMsg` handler anyway, maybe it would be simpler to just do `spanContext<ClientRequestMsg>()`. Just a thought.